.docker/base/Dockerfile

```diff
@@ -4,33 +4,47 @@ LABEL maintainer "Irina Spiridonova <irinasp@microsoft.com>; Darren Gehring <dar
 USER everest
 WORKDIR ${MYHOME}
 
-#PAT token should be provided in the docker run command or as a variable in VSTS build definition 
-ENV PAT $PAT
+ENV CI_LOGS ${MYHOME}"/ci-logs/"
+
+#PAT token and agent name should be provided in the docker run command when docker container is started 
 
 #software-properties-common package is needed for add-apt-repository
-#installing latest version of git
+#installing latest version of git and other packages
 RUN sudo apt-get install --yes software-properties-common
 RUN sudo add-apt-repository ppa:git-core/ppa
 RUN sudo apt-get update
-RUN sudo apt-get install git nuget
+RUN sudo apt-get install --yes \
+  git \
+  nuget \
+  openssh-client
+
+#copying the keys, elevating permissions and changing the key's owner for the key
+WORKDIR ${MYHOME}
+USER everest
+RUN mkdir -p ${MYHOME}/.ssh
+RUN chown everest ${MYHOME}/.ssh
+RUN chmod 700 ${MYHOME}/.ssh
+ADD id_rsa ${MYHOME}/.ssh/id_rsa
+ADD id_rsa.pub ${MYHOME}/.ssh/id_rsa.pub
+RUN sudo chmod 600 ${MYHOME}/.ssh/id_rsa; sudo chmod 600 ${MYHOME}/.ssh/id_rsa.pub 
+RUN sudo chown everest ${MYHOME}/.ssh/id_rsa; sudo chown everest ${MYHOME}/.ssh/id_rsa.pub
 
-#Prepare everest-ci and everest-logs repositories
-RUN git clone https://$PAT@github.com/project-everest/everest-ci.git
-RUN git clone https://$PAT@github.com/project-everest/ci-logs.git
-RUN echo "export CI_LOGS=/$MYHOME/everest/ci-logs/" >> .bashrc 
-RUN . $MYHOME/.bashrc
+#Workaround for cloning a repo in a non-interactive way
+RUN echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ${MYHOME}/.ssh/config
+
+#Prepare everest-ci and ci-logs repositories, calling add-repo.sh script
+ADD add-repo.sh ${MYHOME}/add-repo.sh
+RUN ./add-repo.sh
```

Comment:
Cloning Everest-ci and ci-logs repository instructions is added back to dockerfile. It seems that specifying workdir and building the image from scratch resolved the original problem